### PR TITLE
Refactor object detection detect

### PR DESCRIPTION
**Pull request by rola93**
- avoid convertion to PIL image that was inmediatly converted back to array
- update doc string: when images are arrays they are expected to be BGR (or at least that was what I understood)
- small pythonic changes (remove redundant parentheses, use enumerate instead of counter, etc)

### CODE REVIEW

**Comment by rola93** [comment]
Hey @Borda I've seen that you added CI on #209 maybe you can helpme to understand which test is failing on my last commit....

Thanks in advance!

**Comment by Borda** [comment]
see the following line https://travis-ci.com/OlafenwaMoses/ImageAI/jobs/229109673#L998

**Comment by rola93** [comment]
Thanks for your answer, @Borda 
I have two questions:
1) how could I found it by my self? sorry if it's a newby question, just to avoid asking you every time :smile: 
2) with respect to the error itself: It makes a lot of sence since I changed a little bit the way that results folder is defined.

on current version `image_output` is supposed to be the path to the result image, and must be the result filename, lets  say `image_output.jpg`, and then, objects are stored at `image_output.jpg-objects/foo_object.jpg`. My intention was:
1st) check that the `image_output` parameter given, actually ends with jpg, png or some valid format, otherwise it will fail only after a too long processing/classification time. Another possible option for this point is, instead of failing, complete it with a supported file type (force it to end with .jpg for instance when no supported file type is supported).

2nd) rename the object folder to remove the extension filename. For me at least, it is strange to have a filename that contains `.jpg` or `.tif` on its name. This is not a must, but just makes more sence for me at least. There are two possibilities here: reert this 2nd change and just let strange folder names or update the test (which are failing because of this change). What looks better for you?

Thanks for reading and hope to read from you

**Comment by Borda** [comment]
@rola93  see the `.travis.yaml` file and basically copy-paste to console is a replication of what Travis does... most of the steps you need to do just once like download the data and instal requirements, the testing should be launched every time...
I would recommend to use a debug tool for running the test such as PyCharm and see what is exactly content of your variables `image_output` and if such folder really exists... https://www.jetbrains.com/help/pycharm/performing-tests.html
In my opinion, is that the testing and CI has many weak spots and in the end, there can be even bug in code that you did not touch... :D as I said earlier I tried to help but they refused 8-)

**Comment by OlafenwaMoses** [comment]
@rola93 

As per the PR by @Borda , I fully remember that most of his PR were approved. However, 

- I cautioned him against too many changes out side the scope of the main reason he is making the PR.
- The PR in question #209 does have way to many breaking changes, for which I can't all the time to proper review it. 
- His CI is definitely professional, but the test suite coverage was pretty minimal and I decided to take this myself. 
- I fully encourage contribution to this project but with only the necessary changes. If you desire to make many changes, keep them in different PRs for convenience and separation of concerns.

**Comment by Borda** [comment]
Is coverage minimal? :open_mouth: in that time ImageAI did not have any valid test so what did you expect ... sounds like blaming me for others fault :-D
And another faul from @OlafenwaMoses with many breaking changes! My PR was waiting for couple month when you were inactive, then instead of taking care of contributors you introduced many breaking changed to the past package state... I did three times corrections to be aligned with the actual master (that is the extra changes) but again instead of approving the PR you introduces more breaking changes... so it is really not fair blaming me...

**Comment by OlafenwaMoses** [comment]
@Borda , Inasmuch as I admit I only become fully active to attend to PRs and all issues raised, I want you to understand the following on changes to the project:

- The users' feature demands come first and I will prioritize them over anything else, which is why I had to implement the most demanded fixture (from #8 , training object detection) in the limited time available to me
- The priority pipeline for this project is as ordered below
      ~ features
      ~ stability
      ~ documentation
      ~ code standardization

BTW, I do appreciate your work to contribute to this project.

To resolve this issues, I will be adding a contributors guide soon.

**Comment by rola93** [comment]
> To resolve this issues, I will be adding a contributors guide soon.

Sounds like a great idea. I'd also add a kind of road map/priorities for the project.

File: imageai/Detection/Custom/__init__.py [hunk 1]
```diff
@@ -29,7 +28,7 @@ class DetectionModelTrainer:
     """
 
     def __init__(self):
-
+        print('running local version')
```
[the remark this answers is not in the document]
Comment (OlafenwaMoses) [reply imageai/Detection/Custom/__init__.py]: Oh! Thanks very much. I also missed that during the review. It's okay.

**Pull request merged**